### DISCUSSION TITLE
After upgrading the dependencies for @itwinjs-core in package.json, an error occurs during compression with Electron-builder.

### DISCUSSION THREAD

**Discussion by jh9636** [iTwin/itwinjs-core#6985] [Q&A]
Hello, when I was using version 4.33 of the package, I could successfully package it using electron-builder. However, after upgrading my project to version 4.7.6, the packaging process completes successfully, but when I click the executable .exe file, it throws an error.
The error encountered in the executable file (exe) is illustrated in the image below：
![image](https://github.com/user-attachments/assets/5b5a18fa-e5e3-40f6-972e-4a070f73dfc9)

A screenshot showing the version and dependencies of @opentelemetry/api in the project:
![image](https://github.com/user-attachments/assets/8382d5f4-6182-4318-9aa0-c71cad53f699)

"I'm not quite sure where the issue lies, but if I manually install npm i @opentelemetry/api, I am unable to pack my desktop application successfully. My test starter program is located at https://github.com/iTwin/appui/blob/master/test-apps/appui-test-app/standalone."

After upgrading the version, there are some SCSS warnings when running 'npm run build'. I'm not sure if I'm the only one encountering this issue.

The warning information is shown below:
![image](https://github.com/user-attachments/assets/7ff255cc-6c49-42de-92bb-3c6493158d3f)

On the itwin.js official website, are there any recommended packaging methods? Does the latest version of the itwin application support packaging with electron-builder? I truly appreciate you taking the time to read this. Should you have any suggestions, I would be most grateful.

The following are the versions of the various packages utilized within my project：
```js
PS E:\newBimVersion> npm list     
appui-standalone-app@0.0.0 E:\newBimVersion
├── @axe-core/react@4.3.1
├── @bentley/icons-generic-webfont@1.0.34
├── @bentley/icons-generic@1.0.34
├── @bentley/react-scripts@5.0.7
├── @itwin/appui-abstract@4.7.6
├── @itwin/appui-layout-react@4.8.3
├── @itwin/appui-react@4.15.3
├── @itwin/build-tools@4.7.6
├── @itwin/components-react@4.15.3
├── @itwin/core-backend@4.7.6
├── @itwin/core-bentley@4.7.6
├── @itwin/core-common@4.7.6
├── @itwin/core-electron@4.7.6
├── @itwin/core-frontend@4.7.6
├── @itwin/core-geometry@4.7.6
├── @itwin/core-i18n@4.7.6
├── @itwin/core-mobile@4.7.6
├── @itwin/core-orbitgt@4.7.6
├── @itwin/core-quantity@4.7.6
├── @itwin/core-react@4.15.3
├── @itwin/ecschema-metadata@4.7.6
├── @itwin/ecschema-rpcinterface-common@4.7.6
├── @itwin/ecschema-rpcinterface-impl@4.7.6
├── @itwin/editor-backend@4.7.6
├── @itwin/electron-authorization@0.19.5
├── @itwin/eslint-plugin@4.1.0
├── @itwin/express-server@4.7.6
├── @itwin/frontend-devtools@4.7.6
├── @itwin/hypermodeling-frontend@4.7.6
├── @itwin/imodel-components-react@4.15.3
├── @itwin/imodels-access-backend@5.2.0
├── @itwin/itwinui-icons-color-react@2.1.0
├── @itwin/itwinui-icons-react@2.9.0
├── @itwin/itwinui-react@3.13.1
├── @itwin/itwinui-variables@3.2.0
├── @itwin/map-layers-formats@4.7.6
├── @itwin/map-layers@5.4.4
├── @itwin/presentation-backend@4.7.6
├── @itwin/presentation-common@4.7.6
├── @itwin/presentation-frontend@4.7.6
├── @itwin/presentation-opentelemetry@4.0.2
├── @itwin/reality-data-client@1.2.1
├── @itwin/webgl-compatibility@4.7.6
├── @reduxjs/toolkit@2.2.6
├── @rollup/plugin-babel@6.0.4
├── @types/express-ws@3.0.4
├── @types/express@4.17.21
├── @types/fabric@5.3.7
├── @types/node@18.11.5
├── @types/react-dom@17.0.25
├── @types/react-table@7.7.20
├── @types/three@0.164.1
├── @typescript-eslint/parser@6.21.0
├── antd@5.19.2
├── body-parser@1.20.2
├── classnames@2.3.1
├── cpx2@3.0.2
├── cross-env@5.2.1
├── dotenv-expand@5.1.0
├── dotenv@10.0.0
├── electron-builder@24.13.3
├── electron@24.8.8
├── eslint-config-prettier@8.8.0
├── eslint@8.57.0
├── events@3.3.0
├── express-ws@5.0.2
├── express@4.19.2
├── fabric@5.3.0
├── gltf-import-export@1.0.22
├── npm-run-all@4.1.5
├── null-loader@4.0.1
├── react-beautiful-dnd@13.1.1
├── react-dom@17.0.2
├── react-redux@7.2.9
├── react-table@7.8.0
├── react@17.0.2
├── redux@4.2.1
├── rimraf@3.0.2
├── serve@14.2.3
├── three-mesh-bvh@0.7.6
├── three@0.164.1
├── ts-key-enum@2.0.12
├── typescript@4.9.5
└── webpack@5.93.0
```

**Comment by ben-polinsky** [comment]
Hi,

Looking into this. 
@opentelemetry/api should be an optional peer dependency. Is electron-builder the same version between the 4.3.3 app and the attempted 4.7.6 version?

**Reply by jh9636** [reply]
Hello, thank you very much for your response. Both versions of electron-builder are using the same version. In my package.json, I have "electron-builder": "^24.13.3", which is the latest version of electron-builder.

**Reply by ben-polinsky** [reply]
I see the error is coming from  `@itwin/presentation-opentelemetry`, which you have installed. That has a non-optional peer dependency on `@opentelemetry/api`. I didn't see any obvious changes from 4.3.x versions which would make a difference in triggering this error.

**Reply by jh9636** [reply]
Hello, the version of @itwin/presentation-opentelemetry hasn't changed. Is it possible that the issue is caused by core-backend? This was a project dependency that could be packaged normally before.
![image](https://github.com/user-attachments/assets/b19ed54d-506d-4d14-9770-e896d0ea7c74)

**Reply by ben-polinsky** [reply]
Is it possible a different package manager (npm/pnpm etc), version, or config was used between the two versions? NPM has different default behavior for peer dependencies depending on the version. Perhaps in the earlier version of the app, optional peers were installed, but now they are not. 

This may be it as well:
I see @azure/core-tracing@**1.0.0-preview.13** - this version has opentelemetry as a regular dependency, so it's going to be installed: https://www.npmjs.com/package/@azure/core-tracing/v/1.0.0-preview.13?activeTab=dependencies

In new versions of core-tracing, there is no dependency on opentelemetry: https://www.npmjs.com/package/@azure/core-tracing?activeTab=dependencies

At any rate, since this is a peer dependency (and not optional in presentation-opentelemetry) it is up to you to install.
